webapp/channels/src/components/announcement_bar/text_dismissable_bar.tsx

```diff
@@ -20,67 +20,57 @@ interface Props extends Partial<AnnouncementBarProps> {
-        super(props);
-
-        this.state = {
-            dismissed: true,
-        };
-    }
+const TextDismissableBar = (props: Props) => {
+    const {allowDismissal, text, ...extraProps} = props;
+    const [dismissed, setDismissed] = useState<boolean>(false);
 
-    static getDerivedStateFromProps(props: Props) {
-        const dismissed = localStorage.getItem(localStoragePrefix + props.text?.toString());
-        return {
-            dismissed: (dismissed === 'true'),
-        };
-    }
+    useEffect(() => {
+        const dismissedFromStorage = localStorage.getItem(localStoragePrefix + text?.toString());
+        if (dismissedFromStorage === 'true') {
+            setDismissed(true);
+        }
+    }, [text]);
 
-    handleDismiss = () => {
-        if (!this.props.allowDismissal) {
+    const handleDismiss = () => {
+        if (!allowDismissal) {
             return;
         }
         trackEvent('signup', 'click_dismiss_bar');
 
-        localStorage.setItem(localStoragePrefix + this.props.text?.toString(), 'true');
-        this.setState({
-            dismissed: true,
-        });
-        if (this.props.onDismissal) {
-            this.props.onDismissal();
+        localStorage.setItem(localStoragePrefix + text?.toString(), 'true');
+        setDismissed(
+            true,
+        );
+        if (props.onDismissal) {
+            props.onDismissal();
         }
     };
 
-    render() {
-        if (this.state.dismissed) {
-            return null;
-        }
-        const {allowDismissal, text, ...extraProps} = this.props;
-        return (
-            <AnnouncementBar
-                {...extraProps}
-                showCloseButton={allowDismissal}
-                handleClose={this.handleDismiss}
-                message={
-                    <>
-                        <i className='icon icon-information-outline'/>
-                        {typeof text === 'string' ? (
-                            <Markdown
-                                message={text}
-                                options={{
-                                    singleline: true,
-                                    mentionHighlight: false,
-                                }}
-                            />
-                        ) : text}
-                    </>
-                }
-            />
-        );
+    if (dismissed) {
+        return null;
     }
-}
 
+    return (
+        <AnnouncementBar
+            {...extraProps}
+            showCloseButton={allowDismissal}
+            handleClose={handleDismiss}
+            message={
+                <>
+                    <i className='icon icon-information-outline'/>
+                    {typeof text === 'string' ? (
+                        <Markdown
+                            message={text}
+                            options={{
+                                singleline: true,
+                                mentionHighlight: false,
+                            }}
```

Comment:
Put these in a constant outside of the component, so it doesn't send a new object on every render.